File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        const int n = nums.size();

        if(n < 1) return 0;

        set<int> s(nums.begin(), nums.end());

        int mx = 1, cur = 1;
        auto prev = s.begin();

        for(auto it = next(s.begin()); it != s.end(); it++){
            if(*prev + 1 == * it) cur++;
            else
                mx = max(mx,cur), cur = 1; 
            prev = it;
        }
        
        return max(mx, cur);
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        d = defaultdict(int)

        mx = 0
        for n in nums:
            if not d[n]:
                d[n] = d[n-1] + d[n+1] + 1
                d[n-d[n-1]] = d[n]
                d[n+d[n+1]] = d[n]


                mx = max(mx, d[n])

        return mx